### PR TITLE
[BEAM-646] Use Surgery in the Dataflow Runner

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---
Couple of cleanups and underlying PRs still have to be submitted prior to actual review.

I do want to use this PR to run some RunnableOnService Tests, though.

### CODE REVIEW

**Comment by tgroh** [comment]
Run Dataflow RunnableOnService

**Comment by tgroh** [comment]
Run Dataflow RunnableOnService

**Comment by tgroh** [comment]
Run Dataflow RunnableOnService

**Comment by tgroh** [comment]
R: @kennknowles 

This works now, less a RAT failure unrelated to the change (or at minimum `WordCount` does in both modes.)

**Comment by tgroh** [comment]
Run Dataflow RunnableOnService

**Comment by kennknowles** [comment]
Code looks good, but I want to see what happens between now and green ROS. There's a possibility you'll need a new worker image.

**Comment by tgroh** [comment]
Run Dataflow RunnableOnService

**Comment by tgroh** [comment]
Run Dataflow RunnableOnService

**Comment by tgroh** [comment]
@kennknowles PTAL (Dataflow ROS is green, woop woop)

File: runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslatorTest.java [hunk 1]
```diff
@@ -440,6 +441,7 @@ public void testDiskSizeGbConfig() throws IOException {
   }
 
   @Test
+  @Ignore("TODO: Reenable when the hierarchy is copied")
```
Comment (kennknowles) [inline runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslatorTest.java]: For each TODO: I think some are fixed, some should be just deleted, and the rest should have JIRAs if possible.

**Comment by kennknowles** [comment]
LGTM pending changes just to `@Ignore` unit tests, which shouldn't materially change this PR.

File: runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslatorTest.java (hunk 1 above)
Comment (tgroh) [inline runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslatorTest.java]: All are fixed, with exposure of `performSurgery`

**Comment by kennknowles** [comment]
Run Dataflow RunnableOnService

**Pull request closed without merging**